crates/bevy_remote/src/lib.rs

```diff
@@ -0,0 +1,849 @@
+        self
+    }
+}
+
+impl Default for RemotePlugin {
+    fn default() -> Self {
+        Self::empty()
+            .with_method(
+                "bevy/get".to_owned(),
+                builtin_methods::process_remote_get_request,
+            )
+            .with_method(
+                "bevy/query".to_owned(),
+                builtin_methods::process_remote_query_request,
+            )
+            .with_method(
+                "bevy/spawn".to_owned(),
+                builtin_methods::process_remote_spawn_request,
+            )
+            .with_method(
+                "bevy/insert".to_owned(),
+                builtin_methods::process_remote_insert_request,
+            )
+            .with_method(
+                "bevy/remove".to_owned(),
+                builtin_methods::process_remote_remove_request,
+            )
+            .with_method(
+                "bevy/destroy".to_owned(),
+                builtin_methods::process_remote_destroy_request,
+            )
+            .with_method(
+                "bevy/reparent".to_owned(),
+                builtin_methods::process_remote_reparent_request,
+            )
+            .with_method(
+                "bevy/list".to_owned(),
+                builtin_methods::process_remote_list_request,
+            )
+    }
+}
+
+impl Plugin for RemotePlugin {
+    fn build(&self, app: &mut App) {
+        let mut remote_methods = RemoteMethods::new();
+        let plugin_methods = &mut *self.methods.write().unwrap();
+        for (name, system) in plugin_methods.drain(..) {
+            remote_methods.insert(
+                name,
+                app.main_mut().world_mut().register_boxed_system(system),
+            );
+        }
+
+        app.insert_resource(HostAddress(self.address))
+            .insert_resource(HostPort(self.port))
+            .insert_resource(remote_methods)
+            .add_systems(Startup, start_server)
+            .add_systems(Update, process_remote_requests);
+    }
+}
+
+impl RemoteMethods {
+    /// Creates a new [`RemoteMethods`] resource with no methods registered in it.
+    pub fn new() -> Self {
+        default()
+    }
+
+    /// Adds a new method, replacing any existing method with that name.
+    ///
+    /// If there was an existing method with that name, returns its handler.
+    pub fn insert(
+        &mut self,
+        method_name: impl Into<String>,
+        handler: RemoteMethod,
+    ) -> Option<RemoteMethod> {
+        self.0.insert(method_name.into(), handler)
+    }
+}
+
+/// A system that starts up the Bevy Remote Protocol server.
+fn start_server(mut commands: Commands, address: Res<HostAddress>, remote_port: Res<HostPort>) {
+    // Create the channel and the mailbox.
+    let (request_sender, request_receiver) = channel::bounded(CHANNEL_SIZE);
+    commands.insert_resource(BrpMailbox(request_receiver));
+
+    IoTaskPool::get()
+        .spawn(server_main(address.0, remote_port.0, request_sender))
+        .detach();
```

Comment:
What happens if someone mutates the `HostPort` or `HostAddress` resources? Do we not have a way to kill this task if we want to restart the server on a new port/address?

Reply:
Nope! At the moment the intention is that these resources are populated once with the data from the `Plugin` and are never supposed to be changed. Making it so the server can be restarted might be nice, though.

Reply:
Gotcha! Maybe until then we add doc comments on those resources saying that changing them currently does nothing and that they shouldn't be modified?